### PR TITLE
[#IR-292] issue is completed for review including refresh anf autorefresh functionality

### PR DESCRIPTION
**Pull request by chahat-arora**

### CODE REVIEW

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+  getActivityLog(isRefresh = false, isAutoRefresh = false) {
+    if (!isAutoRefresh) {
+      this.onRefreshStart.emit(true);
+  }
+      return this.retrospectiveService.getActivityLog(this.retrospectiveID, this.sprintID, isAutoRefresh)
+      .takeUntil(this.destroy$)
+      .subscribe(
+              response => {
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts]: Incorrect Indentation!!

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  @Input() retrospectiveID;
+  @Input() sprintID;
+  @Input() isTabActive;
+  @Input() enableRefresh: boolean;
+  @Input() refreshOnChange;
+
+  @Output() onRefreshStart = new EventEmitter<boolean>();
+  @Output() onRefreshEnd = new EventEmitter<boolean>();
+
+  responseData;
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts]: declare the type of the variables...similarly for other places!!

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+                <div *ngIf=isDateChange(i)>
+                    {{trailsDate}}
+                </div>
+            </div>
+            <div>
+                {{data['ActionBy']['FirstName'] |titlecase}} {{ data['ActionBy']['LastName']|titlecase }} {{ data['Action'] }}
+            </div>
+        </mat-list-item>
+    </mat-list>
+</ng-container>
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html]: Add EOF.

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<ng-container>
+    <mat-list class="trails">
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html]: Add custom CSS to highlight the trails and differentiate the action and the doer!!

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        <mat-list-item *ngFor="let data of responseData;let i = index">
+            <div class="trails-date">
+                <div *ngIf=isDateChange(i)>
+                    {{trailsDate}}
+                </div>
+            </div>
+            <div>
+                {{data['ActionBy']['FirstName'] |titlecase}} {{ data['ActionBy']['LastName']|titlecase }} {{ data['Action'] }}
```
Comment (himanshu-garg-165) [inline src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html]: Add consistent spaces around pipes

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html (hunk 4 above)
Comment (chahat-arora) [reply src/app/retrospective/sprint-activity-log/sprint-activity-log.component.html]: Done

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    setInterval(() => {
+      this.date = new Date();
+    }, AUTO_REFRESH_DURATION);
+   }
+
+  ngOnInit() {
+    this.autoRefreshCurrentState = this.enableRefresh;
+
+    Observable.interval(AUTO_REFRESH_DURATION)
+            .takeUntil(this.destroy$)
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts]: Done

File: src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  trailsDate;
+  date: Date = new Date();
+  autoRefreshCurrentState: boolean;
+  private destroy$: Subject<boolean> = new Subject<boolean>();
+
+  constructor(
+    private retrospectiveService: RetrospectiveService,
+    private snackBar: MatSnackBar,
+  ) {
+    setInterval(() => {
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/retrospective/sprint-activity-log/sprint-activity-log.component.ts]: Done

**Pull request merged**